kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExportCodeSpec.kt

```diff
@@ -172,31 +174,48 @@ internal fun ObjCExportCodeSpec.dumpSelectorToSignatureMapping(path: String) {
             is ObjCGetterForNSEnumType -> true
         }
 
-        fun ObjCMethodSpec.getMapping(objcClass: String): String? = when (this) {
+        fun ObjCMethodSpec.getMapping(objcClass: String, overrides: Map<IdSignature?, IdSignature?>): String? = when (this) {
             is ObjCClassMethodForKotlinEnumValuesOrEntries -> "$objcClass.$selector,${valuesFunctionSymbol.signature}"
             is ObjCFactoryMethodForKotlinArrayConstructor -> "$objcClass.${baseMethod.selector},${baseMethod.symbol.signature}"
             is ObjCGetterForKotlinEnumEntry -> "$objcClass.$selector,${irEnumEntrySymbol.signature}"
             is ObjCGetterForObjectInstance -> "$objcClass.$selector,${classSymbol.signature}"
             is ObjCInitMethodForKotlinConstructor -> "$objcClass.${baseMethod.selector},${baseMethod.symbol.signature}"
             is ObjCKotlinThrowableAsErrorMethod -> null
-            is ObjCMethodForKotlinMethod -> "$objcClass.${baseMethod.selector},${baseMethod.symbol.signature}"
+            is ObjCMethodForKotlinMethod -> {
+                val signature = baseMethod.symbol.signature
+                "$objcClass.${baseMethod.selector},${overrides[signature] ?: signature}"
+            }
             is ObjCGetterForNSEnumType -> null
         }
         out.println("\n# Instance methods mapping")
         for (type in types) {
+            val overrides = buildMap {
+                for (method in type.irClassSymbol.descriptor.contributedMethods) {
+                    val signature = symbolTable.descriptorExtension.referenceSimpleFunction(method).signature
```

Comment:
IIUC, instead of referencing the function, it should be possible to do something like
```kotlin
symbolTable.signaturer!!.composeSignature(method)
```
In that case, we don't even need the `symbolTable` here, just the "signaturer".